lib/matplotlib/dviread.py

```diff
@@ -854,47 +881,68 @@ def _register(self, words):
         There is some difference between <foo.pfb and <<bar.pfb in
         subsetting, but I have no example of << in my TeX installation.
         """
-
         # If the map file specifies multiple encodings for a font, we
         # follow pdfTeX in choosing the last one specified. Such
         # entries are probably mistakes but they have occurred.
         # http://tex.stackexchange.com/questions/10826/
         # http://article.gmane.org/gmane.comp.tex.pdftex/4914
 
-        texname, psname = words[:2]
-        effects, encoding, filename = '', None, None
-        for word in words[2:]:
-            if not word.startswith('<'):
-                effects = word
-            else:
-                word = word.lstrip('<')
-                if word.startswith('[') or word.endswith('.enc'):
+        empty_re = re.compile(br'%|\s*$')
+        word_re = re.compile(
+            br'''(?x) (?:
+                 "<\[ (?P<enc1>  [^"]+    )" | # quoted encoding marked by [
+                 "<   (?P<enc2>  [^"]+.enc)" | # quoted encoding, ends in .enc
+                 "<<? (?P<file1> [^"]+    )" | # quoted font file name
+                 "    (?P<eff1>  [^"]+    )" | # quoted effects or font name
+                 <\[  (?P<enc3>  \S+      )  | # encoding marked by [
+                 <    (?P<enc4>  \S+  .enc)  | # encoding, ends in .enc
+                 <<?  (?P<file2> \S+      )  | # font file name
+                      (?P<eff2>  \S+      )    # effects or font name
+            )''')
+        effects_re = re.compile(
+            br'''(?x) (?P<slant> -?[0-9]*(?:\.[0-9]+)) \s* SlantFont
+                    | (?P<extend>-?[0-9]*(?:\.[0-9]+)) \s* ExtendFont''')
+
+        lines = (line.strip()
+                 for line in file
+                 if not empty_re.match(line))
+        for line in lines:
+            effects, encoding, filename = b'', None, None
+            words = word_re.finditer(line)
+
+            w = next(words)
+            texname = w.group('eff2') or w.group('eff1')
+            w = next(words)
+            psname = w.group('eff2') or w.group('eff1')
+
+            for w in words:
+                eff = w.group('eff1') or w.group('eff2')
```

Comment:
The named groups are mutually exclusive so the order doesn't matter for correctness, but there may be a very slight performance difference. I think I was listing the groups in an approximate order of probability of occurrence, e.g. effects are almost certainly quoted because they include arguments, but file and font names are almost certainly not quoted. I can see how this would be confusing; I'll add a comment.